polyfill/lib/ecmascript.mjs

```diff
@@ -1702,7 +1702,7 @@ export const ES = ObjectAssign({}, ES2020, {
     const timeZoneString = timeZone === undefined ? 'Z' : ES.GetOffsetStringFor(outputTimeZone, instant);
     return `${year}-${month}-${day}T${hour}:${minute}${seconds}${timeZoneString}`;
   },
-  TemporalDurationToString: (duration, precision, options = undefined) => {
+  TemporalDurationToString: (duration, precision = 'auto', options = undefined) => {
```

Comment:
why doesn't this match https://tc39.es/proposal-temporal/#sec-temporal-temporaldurationtostring ?

Reply:
Do you mean why isn't the _precision_ argument optional there, or why doesn't it take an options argument? If the latter, it's because the rounding step is done separately (step 6 of https://tc39.es/proposal-temporal/#sec-temporal.duration.prototype.tostring), and no other callers need rounding.

Reply:
no, i mean why is this polyfilled version of the abstract operation taking 3 arguments, when the spec AO takes 11? The eventual implementation in es-abstract will match the spec.

Reply:
This isn't new to @ptomato's change, so not blocking review. But the spec AO could easily also take duration instead of all the duration slot arguments, as the spec AO is only called with fields from duration objects, afaict.

Reply:
Totally reasonable; but if there's going to be a spec change to match the polyfill, the March meeting is pretty close, so it'd be good to get that change in ASAP.

Reply:
In (almost?) all cases, the spec text has you explode out all the fields from Temporal objects and pass those to AOs instead of passing the objects directly, or passing the fields as Records. That was a convention from before I got involved with Temporal, and I just followed it. So I couldn't say for sure why it was chosen, (maybe @pipobscure can?) but I always assumed it was to avoid the hassle of extra "Assert: _duration_ has an [[InitializedTemporalDuration]] internal slot" steps and creating even more Records all over the place. (We do already create a lot of Records for returning fields from AOs.)

I don't really have an opinion on whether this change would be beneficial or not but it seems like it might depend on personal preference.

Reply:
> but if there's going to be a spec change to match the polyfill, the March meeting is pretty close, so it'd be good to get that change in ASAP.

I also think we can consider a polyfill change to match the spec if that's what we'd like - is there any compelling reason to change the spec?

Reply:
i haven't gotten far enough in my review to have an opinion on that yet.

but, this proposal has tended to lead with polyfill changes and update the spec to match, so given that this case is different, it seems like an opportunity to make them the same, in whichever direction.

Reply:
This was just a matter of convenience because it made it easier to wrap my mind around things. No normative intent

Reply:
I think it hasn't really been a goal for this polyfill to match the AOs in the spec exactly, only roughly; but I can see why that would be more important for es-abstract.

In any case, @ljharb what's the resolution you would prefer to see here? I don't think there's a normative change to be made either way, but would you be OK with opening a ticket if you get far enough in your review and decide that your opinion is to change the current approach of exploding out the fields?

Reply:
Oh totally; i don’t think there’s anything actionable here necessarily.

i was pointing out the discrepancy, and was confused by it being intentional since the stated goal of the polyfill was to get feedback on the spec, and if it doesn’t match, it’s not entirely achieving that goal.

Since it’s intentional, and since I’m only reviewing the spec, I’ll certainly open an issue if i think there’s something in the spec that needs changing.

Reply:
Sounds good.

I'd put it a bit differently: the goal of the polyfill is to get feedback from people who _don't_ read the spec :smile: